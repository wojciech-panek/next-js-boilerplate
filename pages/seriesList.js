import React, { PureComponent } from 'react';

export default class SeriesList extends PureComponent {
  render () {
    console.log(this.props);
    return (
      <div>Series List</div>
    )
  }
}
