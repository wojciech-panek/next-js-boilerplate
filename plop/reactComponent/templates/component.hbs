import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { compose } from 'ramda';

import withIntl from '../../shared/lib/withIntl'; // TODO fix relative import
import { Container } from './{{ camelCase name }}.styles';


export class {{ pascalCase name }} extends PureComponent {
  static propTypes = {
  };

  render() {
    return (
      <Container>
        {{ pascalCase name }} component
      </Container>
    );
  }
}

export default compose(
  withIntl,
)({{ pascalCase name }});

