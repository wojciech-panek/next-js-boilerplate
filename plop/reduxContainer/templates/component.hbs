import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';


export class {{ pascalCase name }} extends PureComponent {
  static propTypes = {
  };

  render() {
    return (
      <div>
        {{ pascalCase name }} component
      </div>
    );
  }
}
