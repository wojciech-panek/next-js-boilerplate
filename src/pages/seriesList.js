import React, { PureComponent } from 'react';

export default class SeriesList extends PureComponent {
  render() {
    return (
      <div>Series List</div>
    );
  }
}
