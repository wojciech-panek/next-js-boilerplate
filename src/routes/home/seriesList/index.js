export { default as SeriesList } from './seriesList.container';
