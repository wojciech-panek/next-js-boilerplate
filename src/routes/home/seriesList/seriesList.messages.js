/* eslint-disable max-len */
import { defineMessages } from 'react-intl';


export default defineMessages({
  title: {
    id: 'home.seriesList.title',
    defaultMessage: 'Series List',
  },
  season: {
    id: 'home.seriesList.season',
    defaultMessage: 'Season',
  },
});
