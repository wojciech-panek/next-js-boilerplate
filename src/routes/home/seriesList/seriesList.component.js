import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import { Container, ItemsContainer } from './seriesList.styles';
import messages from './seriesList.messages';
import { H2 } from '../../../shared/theme/typography';
import { Link } from '../../../shared/components/link';


export class SeriesList extends PureComponent {
  static propTypes = {
    items: PropTypes.object.isRequired,
  };

  renderSeriesItem = (item) => {
    return (
      <Link key={item.get('id')} to={`/series/${item.get('id')}`}>
        <a>{item.get('title')} [<FormattedMessage {...messages.season} />: {item.get('season')}]</a>
      </Link>
    );
  };

  render() {
    return (
      <Container>
        <H2>
          <FormattedMessage {...messages.title} />:
        </H2>

        <ItemsContainer>
          {this.props.items.toArray().map(this.renderSeriesItem)}
        </ItemsContainer>
      </Container>
    );
  }
}
