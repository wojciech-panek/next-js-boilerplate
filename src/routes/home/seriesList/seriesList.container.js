import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'ramda';
import { withRouter } from 'next/router';

import withIntl from '../../../shared/lib/withIntl';
import { selectSeriesItems } from '../../../modules/series';
import { SeriesList } from './seriesList.component';


const mapStateToProps = createStructuredSelector({
  items: selectSeriesItems,
});

export const mapDispatchToProps = (dispatch) => bindActionCreators({
}, dispatch);

export default compose(
  withIntl,
  withRouter,
  connect(mapStateToProps, mapDispatchToProps),
)(SeriesList);
