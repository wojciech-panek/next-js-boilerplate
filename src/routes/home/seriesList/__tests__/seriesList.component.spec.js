import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';

import { SeriesList } from '../seriesList.component';


describe('SeriesList: Component', () => {
  const defaultProps = {
    items: fromJS([
      { id: 1, title: 'series 1', season: 1 },
      { id: 2, title: 'series 2', season: 1 },
      { id: 3, title: 'series 3', season: 1 },
    ]),
  };

  const component = (props) => (
    <SeriesList {...defaultProps} {...props} />
  );

  const render = (props = {}) => shallow(component(props));

  it('should render correctly', () => {
    const wrapper = render();
    global.expect(wrapper).toMatchSnapshot();
  });
});
