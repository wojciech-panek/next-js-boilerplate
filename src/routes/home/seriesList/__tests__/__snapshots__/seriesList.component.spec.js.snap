// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeriesList: Component should render correctly 1`] = `
<seriesListstyles__Container>
  <typography__H2>
    <FormattedMessage
      defaultMessage="Series List"
      id="home.seriesList.title"
      values={Object {}}
    />
    :
  </typography__H2>
  <seriesListstyles__ItemsContainer>
    <InjectIntl(withRouter(Connect(Link)))
      key="1"
      to="/series/1"
    >
      <a>
        series 1
         [
        <FormattedMessage
          defaultMessage="Season"
          id="home.seriesList.season"
          values={Object {}}
        />
        : 
        1
        ]
      </a>
    </InjectIntl(withRouter(Connect(Link)))>
    <InjectIntl(withRouter(Connect(Link)))
      key="2"
      to="/series/2"
    >
      <a>
        series 2
         [
        <FormattedMessage
          defaultMessage="Season"
          id="home.seriesList.season"
          values={Object {}}
        />
        : 
        1
        ]
      </a>
    </InjectIntl(withRouter(Connect(Link)))>
    <InjectIntl(withRouter(Connect(Link)))
      key="3"
      to="/series/3"
    >
      <a>
        series 3
         [
        <FormattedMessage
          defaultMessage="Season"
          id="home.seriesList.season"
          values={Object {}}
        />
        : 
        1
        ]
      </a>
    </InjectIntl(withRouter(Connect(Link)))>
  </seriesListstyles__ItemsContainer>
</seriesListstyles__Container>
`;
